macros/src/main/scala/com/softwaremill/macwire/dependencyLookup/ValuesOfTypeInEnclosingClassFinder.scala

```diff
@@ -102,6 +229,16 @@ private[dependencyLookup] class ValuesOfTypeInEnclosingClassFinder[C <: Context]
     }
 
     debug("Looking in the enclosing class/trait")
-    doFind(enclosingClassBody, Nil)
+
+    // Create a special Set that uses structural equality on Tree;
+    // `equals` on trees is reference equality.
+    val emptyTreeSet = TreeSet.empty(new Ordering[Tree] {
+      override def compare(x: c.universe.Tree, y: c.universe.Tree): Int = {
+        if (x.equalsStructure(y)) 0
+        else x.hashCode().compareTo(y.hashCode()) // some arbitrary comparison that should retain the compare semantic
```

Comment:
Ok you're right. There's a tiny little chance that `System.identityHashCode` returns the same value for 2 structurally unequal `Tree`s.
What do you think of this then:
```scala
if (x.equalsStructure(y)) 0
else { 
   val hashCodeCompare = x.hashCode().compareTo(y.hashCode())
   if (hashCodeCompare == 0) -1 else hashCodeCompare
}
```